### PR TITLE
Package changes

### PR DESCRIPTION
**Pull request by SpiritSaint**
This must not merged and only can be a communication place.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "mostafaznv/nova-map-field",
+    "name": "spiritsaint/nova-map-field",
     "description": "Map Field for Laravel Nova",
     "keywords": [
```
Comment (SpiritSaint) [inline composer.json:4]: This change must be ignore and was made just to make a test.

File: composer.json [hunk 2]
```diff
@@ -16,7 +16,6 @@
     "license": "MIT",
     "require": {
         "illuminate/support": "^8.0|^9.0",
-        "laravel/nova": "^4.0",
         "matanyadaev/laravel-eloquent-spatial": "^2.0.1"
```
Comment (SpiritSaint) [inline composer.json:19]: I remove this package because can't deploy a local environment without that.

File: package.json [hunk 3]
```diff
@@ -7,16 +7,20 @@
         "watch-poll": "mix watch --watch-options-poll=1000",
         "hot": "mix watch --hot",
         "prod": "npm run production",
-        "production": "mix --production"
+        "production": "mix --production",
+        "nova:install": "npm --prefix='../../vendor/laravel/nova' ci"
     },
```
Comment (SpiritSaint) [inline package.json:12]: That was added based on nova 4 documentation.

https://nova.laravel.com/docs/4.0/customization/frontend.html#using-nova-mixins

File: package.json [hunk 4]
```diff
@@ -7,16 +7,20 @@
...
+        "form-backend-validation": "^2.3.3",
         "laravel-mix": "^6.0.49",
+        "laravel-nova": "*",
+        "node-sass": "^7.0.3",
         "ol-geocoder": "^4.1.2",
         "resolve-url-loader": "^5.0.0",
         "sass": "^1.52.1",
-        "sass-loader": "^13.0.0",
+        "sass-loader": "^13.1.0",
         "vue": "^3.2.37",
```
Comment (SpiritSaint) [inline package.json:24]: Those packages allows me to compile this package.

File: resources/js/components/form-fields/MultiPolygonFormField.vue [hunk 5]
```diff
@@ -10,7 +10,7 @@
             />
 
             <ol-tile-layer>
-                <ol-source-osm />
+                <ol-source-osm :url="'https://stamen-tiles.a.ssl.fastly.net/toner/{z}/{x}/{y}.png'" />
             </ol-tile-layer>
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/MultiPolygonFormField.vue:15]: This parameter solves the custom tile.

File: resources/js/components/form-fields/PolygonFormField.vue [hunk 6]
```diff
@@ -33,21 +33,28 @@
...
                 </ol-style>
             </ol-vector-layer>
 
             <ol-interaction-select @select="featureSelected" :condition="selectCondition" :features="selectedFeatures">
                 <ol-style>
-                    <ol-style-stroke :color="'red'" :width="2"></ol-style-stroke>
-                    <ol-style-fill :color="`rgba(255, 0, 0, 0.4)`"></ol-style-fill>
+                    <ol-style-stroke :color="`rgba(35, 115, 132, 1)`" :width="2"></ol-style-stroke>
+                    <ol-style-fill :color="`rgba(83, 178, 199, 0.3)`"></ol-style-fill>
                 </ol-style>
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/PolygonFormField.vue:48]: With this i can customize the polygon colors.

File: resources/js/components/form-fields/PolygonFormField.vue [hunk 7]
```diff
@@ -33,21 +33,28 @@
...
             </ol-interaction-select>
 
+            <button type="button"
+                    @click="clear"
+                    class="shadow relative bg-primary-500 hover:bg-primary-400 text-white dark:text-gray-900 cursor-pointer rounded text-sm font-bold focus:outline-none focus:ring ring-primary-200 dark:ring-gray-600 inline-flex items-center justify-center h-9 px-3 shadow relative bg-primary-500 hover:bg-primary-400 text-white dark:text-gray-900"
+                    v-if="isEditable || drawIsEnabled">
+                Borrar Todo
+            </button>
+
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/PolygonFormField.vue:57]: This button will the polygon so i can create a new one.

File: resources/js/components/form-fields/PolygonFormField.vue [hunk 8]
```diff
@@ -67,16 +74,25 @@ export default {
...
     props: ['resourceName', 'resourceId', 'field', 'readonly'],
     methods: {
+        clear() {
+            this.selectedFeatures.pop();
+            this.zones = [];
+            this.drawIsEnabled = true;
+            this.modifyIsEnabled = false;
+            this.setDirty();
+        },
         initCenter() {
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/PolygonFormField.vue:84]: This clears the features and allows me create a new polygon. I don't know if setDirty must be call or not.

File: resources/js/components/form-fields/PolygonFormField.vue [hunk 9]
```diff
@@ -67,16 +74,25 @@ export default {
...
             let value = []
 
             if (this.field.value) {
                 value = JSON.parse(this.field.value)
+
+                console.log(value);
             }
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/PolygonFormField.vue:91]: A dumb console log :rofl:

File: resources/js/components/form-fields/PolygonFormField.vue [hunk 10]
```diff
@@ -67,16 +74,25 @@ export default {
...
 
             if (value.length) {
                 const coordinates = [
-                    value[0].map(v => fromLonLat([v[1], v[0]]))
+                    value[0].map(v => fromLonLat([v[0], v[1]]))
                 ]
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/PolygonFormField.vue:96]: This solves my problem of latitude and longitude order. 

When i create the shape with your package creates me a polygon with inverse latitude and longitude. 

IDK why.

File: resources/js/components/form-fields/PolygonFormField.vue [hunk 11]
```diff
@@ -110,6 +127,8 @@ export default {
         initZones() {
             this.zones = new GeoJSON().readFeatures(this.geoJsonObject)
 
+            console.log(this.zones, this.geoJsonObject)
+
```
Comment (SpiritSaint) [inline resources/js/components/form-fields/PolygonFormField.vue:131]: Another dumb console log.

File: resources/scss/field.scss [hunk 12]
```diff
@@ -9,6 +13,18 @@
...
+}
+
+.ol-control button {
+    background-color: #53B2C7;
+}
+
+.ol-control button:hover {
+    background-color: #53B2C7;
+}
+
```
Comment (SpiritSaint) [inline resources/scss/field.scss:27]: Some UI improvements.

File: src/Fields/MapPolygonField.php [hunk 13]
```diff
@@ -28,7 +28,7 @@ protected function fillAttributeFromRequest(NovaRequest $request, $requestAttrib
                 $points = [];
 
                 foreach ($coordinates as $coordinate) {
-                    $points[] = new Point($coordinate[0], $coordinate[1]);
+                    $points[] = new Point($coordinate[1], $coordinate[0]);
                 }
```
Comment (SpiritSaint) [inline src/Fields/MapPolygonField.php:32]: This solves the problem the inverse location on the backend.

File: webpack.mix.js [hunk 14]
```diff
@@ -16,7 +16,7 @@ mix.setPublicPath('dist')
     .alias({
         'laravel-nova': path.join(
             __dirname,
-            '../../../vendor/laravel/nova/resources/js/mixins/packages.js'
+            '../../vendor/laravel/nova/resources/js/mixins/packages.js'
         )
```
Comment (SpiritSaint) [inline webpack.mix.js:20]: That was changed because i have nova-components folder on root path.

**Review by SpiritSaint — COMMENTED** [review]
@mostafaznv 

This is my humble PR to see what i'm trying to do and how i can do that.

File: resources/js/components/form-fields/PolygonFormField.vue (hunk 6 above)
Comment (mostafaznv) [reply resources/js/components/form-fields/PolygonFormField.vue:48]: I will add some properties to config file. so you can customize colors everywhere.

File: resources/js/components/form-fields/MultiPolygonFormField.vue (hunk 5 above)
Comment (mostafaznv) [reply resources/js/components/form-fields/MultiPolygonFormField.vue:15]: added to config file.
please update the package to v1.7.3

**Pull request closed without merging**